File: apps/minecraft/spigot/src/main/java/crabcraft/net/crabUtilities/update/UpdateExceptions.java
package crabcraft.net.crabUtilities.update;

import java.io.IOException;

public final class UpdateExceptions {
    private UpdateExceptions() {}

    public static class NoReleaseException extends IOException {
        public NoReleaseException(String msg) { super(msg); }
    }

    public static class RateLimitedException extends IOException {
        public RateLimitedException(String msg) { super(msg); }
    }

    public static class AssetNotFoundException extends IOException {
        public AssetNotFoundException(String msg) { super(msg); }
    }

    public static class ChecksumMismatchException extends IOException {
        public ChecksumMismatchException(String msg) { super(msg); }
    }
}

File: apps/minecraft/spigot/src/main/kotlin/crabcraft/net/crabUtilities/update/UpdateExceptions.kt
package crabcraft.net.crabUtilities.update

import java.io.IOException

class UpdateExceptions private constructor() {
    open class NoReleaseException(message: String) : IOException(message)
    open class RateLimitedException(message: String) : IOException(message)
    open class AssetNotFoundException(message: String) : IOException(message)
    open class ChecksumMismatchException(message: String) : IOException(message)
}
